pinot-common/src/main/java/org/apache/pinot/common/broker/BrokerSelectorUtils.java

```diff
@@ -36,8 +35,9 @@ private BrokerSelectorUtils() {
    * @param brokerData: map holding data for table hosting on brokers.
    * @return list of common brokers hosting all the tables.
    */
-  public static List<String> getTablesCommonBrokers(List<String> tableNames, Map<String, List<String>> brokerData) {
-    List<List<String>> tablesBrokersList = new ArrayList<>();
+  public static List<BrokerInfo> getTablesCommonBrokers(List<String> tableNames,
+      Map<String, List<BrokerInfo>> brokerData) {
+    List<List<BrokerInfo>> tablesBrokersList = new ArrayList<>();
```

Comment:
Multiple tables may be in the same tenant. The better way is to get to the set of tenants from the list of tables and then find the common brokers across these tenants. If there is a way to get to tenants from tables, let's do that.

Reply:
There were two ways to get the common tenants. One way is what you suggested. `PinotQueryResource` implements this option. 
The other method is to get the mapping of table->brokers mapping and find the common brokers. The mapping is read from `BROKER_EXTERNAL_VIEW_PATH` The mapping also takes into consideration liveness of the broker. This method is used in `DynamicBrokerSelector`. 
I discussed this with @Jackie-Jiang and he suggested to use the table->brokers mapping as that is the source of truth. So `brokerData` is populated from this mapping and refreshed whenever there is a change in the liveness of brokers.